core_backend/app/llm_call/parse_input.py

```diff
@@ -122,21 +122,74 @@ async def _identify_language(
     """
     Identifies the language of the question.
     """
-    if not isinstance(response, UserQueryResponseError):
-        identified_lang = await _ask_llm_async(
-            question.query_text,
-            IdentifiedLanguage.get_prompt(),
-            litellm_model=LITELLM_MODEL_LANGUAGE_DETECT,
+    if isinstance(response, UserQueryResponseError):
+        return question, response
+
+    llm_identified_lang = await _ask_llm_async(
+        question=question.query_text,
+        prompt=IdentifiedLanguage.get_prompt(),
+        litellm_model=LITELLM_MODEL_LANGUAGE_DETECT,
+    )
+
+    try:
+        identified_lang = getattr(IdentifiedLanguage, llm_identified_lang)
+    except AttributeError:
+        identified_lang = IdentifiedLanguage.UNSUPPORTED
+
+    question.original_language = identified_lang
+    if question.original_language is not None:
+        response.debug_info["original_language"] = question.original_language.value
+
+    processed_response = _process_identified_language_response(
+        identified_lang,
+        response,
+    )
+
+    return question, processed_response
+
+
+def _process_identified_language_response(
+    identified_language: IdentifiedLanguage,
+    response: UserQueryResponse,
+) -> UserQueryResponse | UserQueryResponseError:
+    """Process the identified language and return the response."""
+    is_language_supported = identified_language not in (
+        IdentifiedLanguage.UNSUPPORTED,
+        IdentifiedLanguage.UNINTELLIGIBLE,
+    )
```

Comment:
/nit can we do this instead for clarity and dynamic-ness

```python
identified_language in IdentifiedLanguage.get_supported_languages()
```

Reply:
Oh yeahhh